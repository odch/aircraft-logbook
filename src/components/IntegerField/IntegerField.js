import React from 'react'
import PropTypes from 'prop-types'
import TextField from '@material-ui/core/TextField'

class IntegerField extends React.Component {
  state = {
    focused: false
  }

  handleChange = e => {
    const value = e.target.value
    if (/^\d*$/.test(value)) {
      const intValue = value ? parseInt(value) : undefined
      if (this.props.onChange) {
        this.props.onChange(intValue)
      }
    }
  }

  handleFocus = () => {
    this.setState({
      focused: true
    })
  }

  handleBlur = () => {
    this.setState({
      focused: false
    })
  }

  handleWheel = e => {
    if (this.state.focused === true) {
      e.preventDefault() // prevent number from being changed by scrolling
    }
  }

  render() {
    const { label, value, cy, margin, fullWidth, error, disabled } = this.props
    return (
      <TextField
        label={label}
        value={value || ''}
        onChange={this.handleChange}
        onFocus={this.handleFocus}
        onBlur={this.handleBlur}
        onWheel={this.handleWheel}
        cy={cy}
        type="number"
        margin={margin}
        fullWidth={fullWidth}
        error={error}
        disabled={disabled}
      />
    )
  }
}

IntegerField.propTypes = {
  label: PropTypes.string.isRequired,
  value: PropTypes.number,
  cy: PropTypes.string,
  margin: PropTypes.string,
  fullWidth: PropTypes.bool,
  onChange: PropTypes.func,
  error: PropTypes.bool,
  disabled: PropTypes.bool
}

export default IntegerField
