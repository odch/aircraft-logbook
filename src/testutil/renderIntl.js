import React from 'react'
import renderer from 'react-test-renderer'
import { createMount } from '@material-ui/core/test-utils'
import { IntlProvider } from 'react-intl'
import pretty from 'pretty'
import messages from '../messages'

const LOCALE = 'de'

const renderInIntlProvider = component => (
  <IntlProvider locale={LOCALE} messages={messages[LOCALE]}>
    {component}
  </IntlProvider>
)

const renderIntl = (component, options) =>
  renderer.create(renderInIntlProvider(component), options)

export const renderIntlUpdate = (renderer, component, options) =>
  renderer.update(renderInIntlProvider(component), options)

export const renderIntlMaterial = (component, returnPrettyHtml = false) => {
  const rendered = createMount()(renderInIntlProvider(component))
  return returnPrettyHtml ? pretty(rendered.html()) : rendered
}

export default renderIntl
