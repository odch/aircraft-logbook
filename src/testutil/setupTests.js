import * as enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import moment from 'moment'
import momentTimezone from 'moment-timezone'
import { init as initFeatureToggles } from '../util/featureToggles'

enzyme.configure({ adapter: new Adapter() })

moment.locale('de')
momentTimezone.locale('de')

initFeatureToggles()
