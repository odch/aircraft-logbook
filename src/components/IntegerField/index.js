import IntegerField from './IntegerField'

export default IntegerField
