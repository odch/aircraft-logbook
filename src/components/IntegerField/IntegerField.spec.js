import React from 'react'
import { renderIntlMaterial } from '../../testutil/renderIntl'
import IntegerField from './IntegerField'

describe('components', () => {
  describe('IntegerField', () => {
    it('renders correctly', () => {
      const renderedValue = renderIntlMaterial(
        <IntegerField
          label="test field"
          value={3}
          margin="normal"
          fullWidth
          cy="test-field"
        />,
        true
      )
      expect(renderedValue).toMatchSnapshot()
    })

    it('renders in error mode', () => {
      const renderedValue = renderIntlMaterial(
        <IntegerField
          label="test field"
          value={3}
          margin="normal"
          fullWidth
          cy="test-field"
          error
        />,
        true
      )
      expect(renderedValue).toMatchSnapshot()
    })

    it('renders disabled', () => {
      const renderedValue = renderIntlMaterial(
        <IntegerField
          label="test field"
          value={3}
          margin="normal"
          fullWidth
          cy="test-field"
          disabled
        />,
        true
      )
      expect(renderedValue).toMatchSnapshot()
    })
  })
})
