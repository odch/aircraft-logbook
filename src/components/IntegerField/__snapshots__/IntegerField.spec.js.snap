// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components IntegerField renders correctly 1`] = `
"<div class=\\"MuiFormControl-root-1 MuiFormControl-marginNormal-2 MuiFormControl-fullWidth-4\\" cy=\\"test-field\\"><label class=\\"MuiFormLabel-root-12 MuiFormLabel-filled-16 MuiInputLabel-root-5 MuiInputLabel-formControl-6 MuiInputLabel-animated-9 MuiInputLabel-shrink-8\\" data-shrink=\\"true\\">test field</label>
  <div class=\\"MuiInputBase-root-32 MuiInput-root-19 MuiInput-underline-23 MuiInputBase-fullWidth-41 MuiInput-fullWidth-26 MuiInputBase-formControl-33 MuiInput-formControl-20\\"><input aria-invalid=\\"false\\" class=\\"MuiInputBase-input-42 MuiInput-input-27 MuiInputBase-inputType-45 MuiInput-inputType-30\\" type=\\"number\\" value=\\"3\\"></div>
</div>"
`;

exports[`components IntegerField renders disabled 1`] = `
"<div class=\\"MuiFormControl-root-1 MuiFormControl-marginNormal-2 MuiFormControl-fullWidth-4\\" cy=\\"test-field\\"><label class=\\"MuiFormLabel-root-12 MuiFormLabel-disabled-14 MuiFormLabel-filled-16 MuiInputLabel-root-5 MuiInputLabel-formControl-6 MuiInputLabel-animated-9 MuiInputLabel-shrink-8\\" data-shrink=\\"true\\">test field</label>
  <div class=\\"MuiInputBase-root-32 MuiInput-root-19 MuiInput-underline-23 MuiInputBase-disabled-35 MuiInput-disabled-22 MuiInputBase-fullWidth-41 MuiInput-fullWidth-26 MuiInputBase-formControl-33 MuiInput-formControl-20\\"><input aria-invalid=\\"false\\" class=\\"MuiInputBase-input-42 MuiInput-input-27 MuiInputBase-disabled-35 MuiInput-disabled-22 MuiInputBase-inputType-45 MuiInput-inputType-30\\" disabled=\\"\\" type=\\"number\\" value=\\"3\\"></div>
</div>"
`;

exports[`components IntegerField renders in error mode 1`] = `
"<div class=\\"MuiFormControl-root-1 MuiFormControl-marginNormal-2 MuiFormControl-fullWidth-4\\" cy=\\"test-field\\"><label class=\\"MuiFormLabel-root-12 MuiFormLabel-error-15 MuiFormLabel-filled-16 MuiInputLabel-root-5 MuiInputLabel-formControl-6 MuiInputLabel-animated-9 MuiInputLabel-shrink-8\\" data-shrink=\\"true\\">test field</label>
  <div class=\\"MuiInputBase-root-32 MuiInput-root-19 MuiInput-underline-23 MuiInputBase-error-38 MuiInput-error-24 MuiInputBase-fullWidth-41 MuiInput-fullWidth-26 MuiInputBase-formControl-33 MuiInput-formControl-20\\"><input aria-invalid=\\"true\\" class=\\"MuiInputBase-input-42 MuiInput-input-27 MuiInputBase-inputType-45 MuiInput-inputType-30\\" type=\\"number\\" value=\\"3\\"></div>
</div>"
`;
